daemon/src/maker_cfd_actor.rs

```diff
@@ -185,27 +197,77 @@ pub fn new(
                             cfd,
                         );
 
+                        current_contract_setup = Some(inbox.clone());
+                        for msg in contract_setup_message_buffer.clone() {
```

Comment:
You could drain the buffer here, then you don't need to clone and reset the buffer.